frontend/src/stores/relationshipStore.ts

```diff
@@ -0,0 +1,164 @@
+import { ValidRelationships, type Relationship } from '@/types/Relationship';
+import type { User } from '@/types/User';
+import { getRequest, patchRequest, postRequest } from '@/utils/apiRequests';
+import { defineStore } from 'pinia';
+import { io, type Socket } from 'socket.io-client';
+
+export const useRelationshipStore = defineStore('relationship', {
+	//  actions == data definitions
+	state: () => ({
+		relationships: <Relationship[]>[],
+		me: {} as User,
+		isInitialized: false,
+		socket: {} as Socket,
+	}),
+	// getters == computed values
+	getters: {},
+	// actions == methods
+	actions: {
+		async initialize() {
+			if (this.isInitialized === false) {
+				// for when sockets are setup
+				this.isInitialized = false;
+				this.socket = io('http://localhost:8080/user/relationship');
+				await this.refreshRelationships();
```

Comment:
not sure, but should this be before init of socket? probably doesn't matter that much...